src/db.rs

```diff
@@ -0,0 +1,82 @@
+use sqlx::migrate::{Migrate, MigrateDatabase, Migrator};
+use sqlx::pool::PoolOptions;
+use sqlx::Any;
+use sqlx::Pool;
+use tracing::info;
+
+static MIGRATOR: Migrator = sqlx::migrate!("schemas/database");
+
+pub struct Database {
+    pool: Pool<Any>,
+}
+
+pub enum MigrationStatus {
+    Empty,
+    Dirty,   // the database has an unexpected schema
+    Behind,  // the database is out of date
+    Current, // nothing needs to be done
+    Ahead,   // the binary is out of date
+}
+
+impl Database {
```

Comment:
Please copy over this file as starting point: https://github.com/worldcoin/signup-sequencer/blob/b5fcb296baac47cc0cd66c481da3259c6b5e7b82/src/database.rs#L42

We should probably abstract this out to a utility library at some point, but for now let's manually keep it in sync so this migration will be easier later.

Reply:
I'm happy to add more features if you think something is missing, but then signup-sequencer probably wants them too!

Reply:
in this follow-up PR I port it over to using a copy of the signup-sequencer's logic: https://github.com/worldcoin/tx-sitter/pull/20